lib/shopify_cli/theme/dev_server.rb

```diff
@@ -65,6 +94,10 @@ def setup(ctx, root, host, theme_identifier, port, poll, editor_sync, stable, mo
         @stable = stable
         @mode = mode
         @block = block
+
+        @include_filter = ShopifyCLI::Theme::IncludeFilter.new(root, includes)
+        @ignore_filter = ShopifyCLI::Theme::IgnoreFilter.from_path(root)
+        @ignore_filter.add_patterns(ignores) if ignores
```

Comment:
(optional) Personally, I believe we could keep the pattern of saving the flags as server properties, so instead of creating `include_filter` and `ignore_filter` attributes, we'd have the new `includes` and `ignores`.

And then, I'd defer the instantiation to the memoized method:
```ruby
def ignore_filter
  @ignore_filter ||= IgnoreFilter.from_path(root).tap do |filter|
    filter.add_patterns(ignores) if ignores
  end
end

def include_filter
  @include_filter ||= ShopifyCLI::Theme::IncludeFilter.new(root, includes)
end
```

(this change aims to prevent the `setup` method from growing in complexity)

Reply:
ah that makes a lot of sense 👍